### PR TITLE
documentation/S3C-4377: utapi metrics protection requirements

### PR DESCRIPTION
**Pull request by terrydervaux**
Utapi metrics protection requirements

### CODE REVIEW

File: docs/requirements/utapi-metrics-protection.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+mechanism. However, root users have unlimited access to consumption metrics. For
+instance, `accountA` root user can access `accountB` metrics.
+
+Service provider customers are complaining about the current implementation. They
+would like metrics to be protected at the account level so that they can expose
+UTAPI to their customers.
+
+In the meantime, service providers have used this behavior to implement their
+billing system. So, if we change the current behavior, we need to provide service
+providers another to bill their customers.
```
Comment (jonathan-gramain) [inline docs/requirements/utapi-metrics-protection.md:18]:
```suggestion
providers another way to bill their customers.
```

**Review by jonathan-gramain — APPROVED** [review]

File: docs/requirements/utapi-metrics-protection.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+**In order** to bill storage account owners
+
+Acceptance criteria
+
+* A single set of keys must allow retrieving all storage accounts consumption
+  metrics
+* Credentials must not allow to create, update or delete storage accounts, IAM,
+  S3, or STS resources (e.g: buckets, objects, users, policies, roles…)
+
```
Comment (cmrh) [inline docs/requirements/utapi-metrics-protection.md:36]: If i understand this correctly, we would have the equivalent of a superadmin in vault just for utapi metrics, and account owners would have access to utapi metrics only in their account, like we do today for IAM users to access utapi?

Comment (cmrh) [reply docs/requirements/utapi-metrics-protection.md:36]: To be clearer, we apply IAM policies to the IAM users today to access utapi metrics.

Comment (terrydervaux) [reply docs/requirements/utapi-metrics-protection.md:36]: Yes exactly.

**Review by cmrh — APPROVED** [review]

**Review by tmacro — APPROVED** [review]

**Pull request merged**